src/ranged.cpp

```diff
@@ -593,36 +602,67 @@ void player::fire_gun( const tripoint &targ_arg, bool burst )
     }
 }
 
-void game::throw_item( player &p, const tripoint &target, item &thrown,
-                       std::vector<tripoint> &trajectory )
+dealt_projectile_attack player::throw_item( const tripoint &target, const item &to_throw )
 {
+    // We'll be constructing a projectile
+    // Start by copying the item into it
+    projectile proj;
+    proj.drop = std::unique_ptr<item>( new item( to_throw ) );
```

Comment:
Usually, I use `drop.reset( new item( ... ) )`, but I also that it makes no difference beside being shorter (-: